fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/AddDecodeNodeForDictStringRule.java

```diff
@@ -390,6 +391,10 @@ public OptExpression visitPhysicalOlapScan(OptExpression optExpression, DecodeCo
                     if (!checkColumnCouldApply.getAsBoolean()) {
                         continue;
                     }
+                    if (hasEncodedColumnRefs.contains(stringColumn)) {
```

Comment:
why one table could got multi same string id?

Reply:
This only happens for MV rewrite for now. For example:
- 

MV:
```
      select
        sum(ps_supplycost * ps_availqty) * 0.0001000000
      from
        partsupp,
        supplier,
        nation
      where
        ps_suppkey = s_suppkey
        and s_nationkey = n_nationkey
     group by n_name
```

Query:
```
      select
        sum(ps_supplycost * ps_availqty) * 0.0001000000
      from
        partsupp,
        supplier,
        nation
      where
        ps_suppkey = s_suppkey
        and s_nationkey = n_nationkey
        and n_name = 'GERMANY'

union all

      select
        sum(ps_supplycost * ps_availqty) * 0.0001000000
      from
        partsupp,
        supplier,
        nation
      where
        ps_suppkey = s_suppkey
        and s_nationkey = n_nationkey
        and n_name = 'GERMANY'
```

Original query will be rewritten by the MV by replacing original Join+Agg node to ScanNode, the `ScanNode` of MV will have the same column-ref for the output (and the same `stringColumn`).

Reply:
I didn't specify exactly what kind of plan it would be rewritten to, but it doesn't seem like an unreasonable reason.
If there are multiple scans, the output id of the string should not be the same.

Reply:
Yep, this is a problem. If the two MV's ScanNode existed in the same fragment, it may cause other bugs.

Can we use this pr as a protect for the same output ids? I will avoid the repeated slot-ids for the mv rewrite later.